blueoil/cmd/train.py

```diff
@@ -13,39 +13,378 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # =============================================================================
-import os
 from datetime import datetime
+import math
+import os
 
-from tensorflow.io import gfile
+import click
+import tensorflow as tf
+from tensorflow.core.util.event_pb2 import SessionLog
+from tensorflow.keras.utils import Progbar
 import yaml
 
-from executor.train import run as run_train
+from blueoil import environment
+from blueoil.common import Tasks
+from blueoil.datasets.base import ObjectDetectionBase
+from blueoil.datasets.dataset_iterator import DatasetIterator
+from blueoil.datasets.tfds import TFDSClassification, TFDSObjectDetection
+from blueoil.utils import config as config_util
+from blueoil.utils import executor
 from blueoil.utils import horovod as horovod_util
+from blueoil.utils import module_loader
+
+
+def _save_checkpoint(saver, sess, global_step, step):
+    checkpoint_file = "save.ckpt"
+    saver.save(
+        sess,
+        os.path.join(environment.CHECKPOINTS_DIR, checkpoint_file),
+        global_step=global_step,
+    )
+
+
+def setup_dataset(config, subset, rank):
+    DatasetClass = config.DATASET_CLASS
+    dataset_kwargs = {key.lower(): val for key, val in config.DATASET.items()}
+
+    # If there is a settings for TFDS, TFDS dataset class will be used.
+    tfds_kwargs = dataset_kwargs.pop("tfds_kwargs", {})
+    if tfds_kwargs:
+        if issubclass(DatasetClass, ObjectDetectionBase):
+            DatasetClass = TFDSObjectDetection
+        else:
+            DatasetClass = TFDSClassification
+
+    dataset = DatasetClass(subset=subset, **dataset_kwargs, **tfds_kwargs)
+    enable_prefetch = dataset_kwargs.pop("enable_prefetch", False)
+    return DatasetIterator(dataset, seed=rank, enable_prefetch=enable_prefetch)
+
+
+def start_training(config):
```

Comment:
This function is too long, But it's copied from original `train.py`.
We will refactor in other PR.